### PR TITLE
[Oracle] Allow unobfuscated plan logging

### PR DESCRIPTION
**Pull request by nenadnoveljic**
### What does this PR do?

Adding the undocumented parameter `log_unobfuscated_plans` to allow logging of unobfuscated plan information.

https://datadoghq.atlassian.net/browse/DBMON-2788

### Motivation

This is for debugging purposes if a customer hits an obfuscated error.

### Describe how to test/QA your changes

n/a

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/collector/corechecks/oracle-dbm/statements.go [hunk 1]
```diff
@@ -809,7 +809,12 @@ func (c *Check) StatementMetrics() (int, error) {
 									if err == nil {
 										stepPayload.AccessPredicates = obfuscated.Query
 									} else {
-										stepPayload.AccessPredicates = "obfuscation error"
+										stepPayload.AccessPredicates = "Access obfuscation error"
+										if c.config.ExecutionPlans.LogUnobfuscatedPlans {
+											stepPayload.AccessPredicates = fmt.Sprintf("%s, unobfuscated filter: %s", stepPayload.AccessPredicates, stepRow.AccessPredicates.String)
+										}
+										log.Errorf(stepPayload.FilterPredicates)
```
Comment (lu-zhengda) [inline pkg/collector/corechecks/oracle-dbm/statements.go:816]: typo? should it be `log.Errorf(stepPayload.AccessPredicates)`?

**Review by maycmlee — APPROVED** [review]
👍 for docs

File: pkg/collector/corechecks/oracle-dbm/statements.go (hunk 1 above)
Comment (nenadnoveljic) [reply pkg/collector/corechecks/oracle-dbm/statements.go]: Yes, thx!. Fixed now.

File: pkg/collector/corechecks/oracle-dbm/statements.go [hunk 2]
```diff
@@ -809,17 +809,23 @@ func (c *Check) StatementMetrics() (int, error) {
 									if err == nil {
 										stepPayload.AccessPredicates = obfuscated.Query
 									} else {
-										stepPayload.AccessPredicates = "obfuscation error"
-										log.Errorf("Access obfuscation error")
+										stepPayload.AccessPredicates = "Access obfuscation error"
+										if c.config.ExecutionPlans.LogUnobfuscatedPlans {
+											stepPayload.AccessPredicates = fmt.Sprintf("%s, unobfuscated filter: %s", stepPayload.AccessPredicates, stepRow.AccessPredicates.String)
+										}
+										log.Errorf(stepPayload.AccessPredicates)
```
[the remark this answers is not in the document]
Comment (nenadnoveljic) [reply pkg/collector/corechecks/oracle-dbm/statements.go]: Corrected now, thx.

**Review by alexandre-normand — APPROVED** [review]

**Review by lu-zhengda — APPROVED** [review]

**Pull request merged**